am-lib/src/main/java/uk/gov/hmcts/reform/amlib/DefaultRoleSetupImportService.java

```diff
@@ -0,0 +1,216 @@
+     */
+    public void addService(@NonNull String serviceName) {
+        if (serviceName.isEmpty()) {
+            throw new IllegalArgumentException(emptyServiceError);
+        }
+
+        jdbi.useExtension(DefaultRoleSetupRepository.class,
+            dao -> dao.addService(serviceName, ""));
+    }
+
+    /**
+     * Creates a new unique service, with a description, or updates if already exists.
+     *
+     * @param serviceName        the name of the service.
+     * @param serviceDescription a description of the service.
+     */
+    public void addService(@NonNull String serviceName, String serviceDescription) {
+        if (serviceName.isEmpty()) {
+            throw new IllegalArgumentException(emptyServiceError);
+        }
+
+        jdbi.useExtension(DefaultRoleSetupRepository.class,
+            dao -> dao.addService(serviceName, serviceDescription));
+    }
+
+    /**
+     * Creates a new unique role or updates if already exists.
+     *
+     * @param roleName               the name of the role.
+     * @param roleType               the type of role.
+     * @param securityClassification the security classification for the role.
+     * @param accessType             the access type for the role.
+     */
+    public void addRole(@NonNull String roleName,
+                        @NonNull RoleType roleType,
+                        @NonNull SecurityClassification securityClassification,
+                        @NonNull AccessType accessType) {
+        if (roleName.isEmpty()) {
+            throw new IllegalArgumentException("Role name cannot be empty");
+        }
+
+        jdbi.useExtension(DefaultRoleSetupRepository.class,
+            dao -> dao.addRole(roleName, roleType, securityClassification, accessType));
+    }
+
+    /**
+     * Creates a new resource definition or updates if already exists.
+     *
+     * @param serviceName  the name of the service the resource belongs to.
+     * @param resourceType the type of resource.
+     * @param resourceName the name of the resource.
+     */
+    public void addResourceDefinition(@NonNull String serviceName,
+                                      @NonNull String resourceType,
+                                      @NonNull String resourceName) {
+        if (serviceName.isEmpty()) {
+            throw new IllegalArgumentException(emptyServiceError);
+        }
+
+        if (resourceType.isEmpty() || resourceName.isEmpty()) {
+            throw new IllegalArgumentException("Resource cannot contain empty values");
+        }
+
+        jdbi.useExtension(DefaultRoleSetupRepository.class, dao ->
+            dao.addResourceDefinition(serviceName, resourceType, resourceName));
+
+    }
+
+    /**
+     * Creates a new resource attribute with default permissions for a role or updates if already exists.
+     *
+     * <p>Operation uses a transaction and will rollback if any errors are encountered whilst adding entries.
+     * {@link PersistenceException}
```

Comment:
What a surprise ;) When I look at that Javadoc I cannot resist thinking that link does not fit there. There is no place for it at the moment. 

What about:

```
...
@throws PersistenceException if any errors were encountered causing transaction rollback
```